### PR TITLE
First pass at connecting to SQS

### PR DESCRIPTION
**Pull request by nodanaonlyzuul**
Howdy partner.

This PR does the bare-minimum to connect to SQS.
It's mostly to setup some infrastructure around configuration variables.

Here are the broad strokes.

1.  It adds a config/settings.yml file to source control.
It's okay to be in source control because it just reads environment variables.

2.  Locally - developers can add the environment variables to `./config/.env`.
In production, we'll add the environment variables directly to whatever we use to serve this.
(Probably Elastic Container Service).

3.  Finally `./consume_messages.rb` is where the environment variables are used to connect to SQS and we simply print, but don't delete, the messages.

Tuesday - I can show you this working on a queue I setup in nypl-sandbox.

I'm going to add a few inline comments to walk you through `consume_messages.rb`

### CODE REVIEW

File: consume_messages.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require 'aws-sdk'
+
+# Bring environment variables in ./config/.env into scope.
```
Comment (nodanaonlyzuul) [inline consume_messages.rb]: We can use the [dotenv gem](https://github.com/bkeepers/dotenv) gem in development as an easy way to
set the environment variables that are used in `settings.yml`.

File: consume_messages.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+response = sqs_client.receive_message(queue_url: settings['sqs_queue_url'], attribute_names: ['All'])
+
+# Each message is an instance of:
+# https://docs.aws.amazon.com/sdkforruby/api/Aws/SQS/Types/Message.html
+response.messages.each do |message|
+  puts message.body
+end
+
+
```
Comment (nodanaonlyzuul) [inline consume_messages.rb]: I've had problems getting long-polling working locally so kept this in, commented-out, for now.
But maybe we can look into that together.

File: consume_messages.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+require 'yaml'
+path_to_settings = File.join(__dir__, "config", "settings.yml")
+settings = YAML.load(ERB.new(File.read(path_to_settings)).result)
+
+# Configure SQS Client
+credentials = Aws::Credentials.new(settings['aws_key'], settings['aws_secret'])
+sqs_client  = Aws::SQS::Client.new(region: 'us-east-1', credentials: credentials)
+
+response = sqs_client.receive_message(queue_url: settings['sqs_queue_url'], attribute_names: ['All'])
+
```
Comment (nodanaonlyzuul) [inline consume_messages.rb]: Here's where we retrieve and iterate over messages from SQS.
There's no guarantee about how many messages come back from `sqs_client.receive_message()`
but I've read online not to expect more than a few unless you have LOTS of messages in your queue.

File: consume_messages.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+# Create a hash that contains the configuarable variables
+require 'erb'
+require 'yaml'
+path_to_settings = File.join(__dir__, "config", "settings.yml")
+settings = YAML.load(ERB.new(File.read(path_to_settings)).result)
+
+# Configure SQS Client
+credentials = Aws::Credentials.new(settings['aws_key'], settings['aws_secret'])
+sqs_client  = Aws::SQS::Client.new(region: 'us-east-1', credentials: credentials)
+
```
Comment (nodanaonlyzuul) [inline consume_messages.rb]: regarding `attribute_names`....

[Attributes](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-attributes.html) are like headers that come with the message.

I experimented with adding a message to the queue via the AWS web interface and I'm 90% sure that the
headers [in this gist](https://gist.github.com/nodanaonlyzuul/479cf45e95668851696a63540ce3f726) are part of a message, automatically, regardless of how the message was created (some Ruby/Python/JS library, or the web console).

**Comment by ktp242** [comment]
Thanks for the PR. Looks good. But I feel I still need you to walk me through so I will be sure to understand every thing .

File: Gemfile [hunk 5]
```diff
@@ -1,6 +1,7 @@
 source 'http://rubygems.org'
```
Comment (ktp242) [inline Gemfile]: I read the documentation of Gemfile. Should we use https?

Comment (nodanaonlyzuul) [reply Gemfile]: Hey @ktp242 

Yes, you're 100% correct about using https, not http.
I'll correct it, thank you.

We don't need a `do` on that line - let's talk about it!

File: consume_messages.rb (hunk 4 above)
Comment (ktp242) [reply consume_messages.rb]: 👍

**Pull request merged**